build-system/tasks/visual-diff/index.js

```diff
@@ -526,86 +536,58 @@ async function snapshotWebpages(browser, webpages) {
-              'to the test page',
+            'The browser test runner failed to complete the navigation to the test page',
             navigationError,
             consoleMessages
           );
           log('warning', 'Continuing to verify page regardless...');
         }
 
-        // Perform visibility checks: wait for all AMP built-in loader dots
-        // to disappear (i.e., all visible components are finished being
-        // layed out and external resources such as images are loaded and
-        // displayed), then, depending on the test configurations, wait for
-        // invisibility/visibility of specific elements that match the
-        // configured CSS selectors.
-        await waitForPageLoad(page, name);
-        if (webpage.loading_incomplete_selectors) {
-          await verifySelectorsInvisible(
-            page,
-            name,
-            webpage.loading_incomplete_selectors
-          );
-        }
-        if (webpage.loading_complete_selectors) {
-          await verifySelectorsVisible(
-            page,
-            name,
-            webpage.loading_complete_selectors
-          );
-        }
-
-        // Based on test configuration, wait for a specific amount of time.
-        if (webpage.loading_complete_delay_ms) {
-          log(
-            'verbose',
-            'Waiting',
-            colors.cyan(`${webpage.loading_complete_delay_ms}ms`),
-            'for loading to complete'
-          );
-          await sleep(webpage.loading_complete_delay_ms);
-        }
-
-        // Run any other custom code located in the test's interactive_tests
-        // file. If there is no interactive test, this defaults to an empty
-        // function.
-        await testFunction(page, name);
-
-        // Execute post-scripts that clean up the page's HTML and send
-        // prepare it for snapshotting on Percy. See comments inside the
-        // snippet files for description of each.
-        await page.evaluate(REMOVE_AMP_SCRIPTS_SNIPPET);
-        await page.evaluate(FREEZE_FORM_VALUE_SNIPPET);
-        await page.evaluate(FREEZE_CANVAS_IMAGE_SNIPPET);
-
-        // Create a default set of snapshot options for Percy and modify
-        // them based on the test's configuration.
-        const snapshotOptions = {};
-        if (webpage.enable_percy_javascript) {
-          snapshotOptions.enableJavaScript = true;
-        }
+        let performSnapshot = true;
+        try {
+          // Perform visibility checks: wait for all AMP built-in loader dots
+          // to disappear (i.e., all visible components are finished being
+          // layed out and external resources such as images are loaded and
+          // displayed), then, depending on the test configurations, wait for
+          // invisibility/visibility of specific elements that match the
+          // configured CSS selectors.
+          await waitForPageLoad(page, name);
```

Comment:
I see this message [in the logs](https://travis-ci.com/github/ampproject/amphtml/jobs/469888376#L2826):

> waitFor is deprecated and will be removed in a future release. See https://github.com/puppeteer/puppeteer/issues/6214 for details and how to migrate your code.

Is there a straightforward way to migrate the usage in this PR?

Reply:
Not so straightforward, I added fixing this to my tasks list though